### PR TITLE
Convert diag_ping.php

### PR DESCRIPTION
**Pull request by sbeaver-netgate**

### CODE REVIEW

File: usr/local/www/diag_ping.php [hunk 1]
```diff
@@ -47,10 +47,20 @@
...
 define('MAX_COUNT', 10);
 define('DEFAULT_COUNT', 3);
 
+function create_sourceaddresslist() {
+	$list = array('any' => 'Any');
+
+	$sourceips = get_possible_traffic_source_addresses(true);
+
+	foreach ($sourceips as $sipvalue => $sipname)
+		$list[$sipname[value]] = $sipname[name];
```
Comment (SjonHortensius) [inline usr/local/www/diag_ping.php]: This is invalid code; please quote the array-keys

File: usr/local/www/diag_ping.php [hunk 2]
```diff
@@ -79,118 +89,102 @@
...
+	'Maximum nuber of pings',
+	$count,
+	array_combine(range(1, MAX_COUNT), range(1, MAX_COUNT))
+))->setHelp('Select the maximum number pings');
+
+$form->add($section);
+print $form;
+
+if($do_ping && !empty($result) && !$input_errors) {
+	print("<div class=\"panel panel-info\">" . "<div class=\"panel-heading\">Results</div>");
```
Comment (SjonHortensius) [inline usr/local/www/diag_ping.php]: Can you use `?><div...` instead (without print/echo)? Also; the `&nbsp;`'s are unwanted; we'll add spacing through css if we need to. Also; please use nl2br($result)

File: usr/local/www/diag_ping.php [hunk 3]
```diff
@@ -79,118 +89,102 @@
...
+	print("&nbsp;&nbsp;" . str_replace("\n", "<br />&nbsp;&nbsp;", $result));
+	print('</div></div>');
+}
+
+include("foot.inc");
+
```
Comment (SjonHortensius) [inline usr/local/www/diag_ping.php]: Please delete the last 3 lines. We don't need the closing tag and whitespace is unneeded as well (including the last enter)

Comment (sbeaver-netgate) [inline usr/local/www/diag_ping.php]: Updated per your suggestions, except that I would like to leave:

```
    // ToDo: Revert to nl2br() and remove &nbsp;when the CSS has been adjusted
print("&nbsp;&nbsp;" . str_replace("\n", "<br />&nbsp;&nbsp;", $result));
```

in place for now. I am asked to demonstrate these pages from time time and I need them to look good even without the planned CSS adjustment :)

Steve

> On Apr 14, 2015, at 1:53 PM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_ping.php https://github.com/SjonHortensius/pfsense/pull/69#discussion_r28351721:
> 
> > -   'Maximum nuber of pings',
> > -   $count,
> > -   array_combine(range(1, MAX_COUNT), range(1, MAX_COUNT))
> >   +))->setHelp('Select the maximum number pings');
> >   +
> >   +$form->add($section);
> >   +print $form;
> >   +
> >   +if($do_ping && !empty($result) && !$input_errors) {
> > -   print("<div class=\"panel panel-info\">" . "<div class=\"panel-heading\">Results</div>");
> > -   print("&nbsp;&nbsp;" . str_replace("\n", "<br />&nbsp;&nbsp;", $result));
> > -   print('</div></div>');
> >   +}
> >   +
> >   +include("foot.inc");
> >   +
> >   Please delete the last 3 lines. We don't need the closing tag and whitespace is unneeded as well (including the last enter)
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/69/files#r28351721.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/

**Pull request merged**